pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -383,6 +387,68 @@ func ensureKubeAPIServerTrustedCA(client coreclientv1.CoreV1Interface, recorder
 	return err
 }
 
+func ensureLocalhostRecoverySAToken(client coreclientv1.CoreV1Interface, recorder events.Recorder) error {
+	requiredSA := resourceread.ReadServiceAccountV1OrDie(v410_00_assets.MustAsset("v4.1.0/kube-apiserver/localhost-recovery-sa.yaml"))
+	requiredToken := resourceread.ReadSecretV1OrDie(v410_00_assets.MustAsset("v4.1.0/kube-apiserver/localhost-recovery-token.yaml"))
+
+	saClient := client.ServiceAccounts(operatorclient.TargetNamespace)
+	serviceAccount, err := saClient.Get(requiredSA.Name, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		serviceAccount, err = saClient.Create(requiredSA)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create custom secret to be populated with SA token so we have a stable name
+	secretsClient := client.Secrets(operatorclient.TargetNamespace)
+	token, err := secretsClient.Get(requiredToken.Name, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		token, err = secretsClient.Create(requiredToken)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Token creation / injection for a SA is asynchronous.
+	// We will report and error if it's missing, go degraded and get re-queued when the SA token is updated.
```

Comment:
when do we go degraded? On operator startup?

Again my upper questoin: why not move the token secret into the manifest dir. Then there is a time gap anyway between creation and code here.

Reply:
That would just change the timing on the race (nor I think moving it there is correct), we would still have to handle the case.

Sounds like currently we go degraded immediately in all the other cases and this goes in line with that pattern, although we should probably adjust it in the future. That would likely be in a common place, so I don't think this PR would be blocked by that.